odoo/modules/module.py

```diff
@@ -249,6 +249,14 @@ def module_manifest(path):
         if os.path.isfile(opj(path, manifest_name)):
             return opj(path, manifest_name)
 
+def read_manifest(addons_path, module):
+    mod_path = opj(addons_path, module)
+    manifest_path = module_manifest(mod_path)
+    if manifest_path:
+        with open(manifest_path, 'rb') as fd:
+            manifest_data = fd.read()
+        return ast.literal_eval(pycompat.to_text(manifest_data))
```

Comment:
there's no point reading the file in binary mode to straight convert it to string afterwards, might as well read in text mode. Most of the code you copied comes from legacy P2/P3 compatibility stuff, it can be modernized/simplified for P3